File: CustomUi/app/src/main/java/com/jwplayer/customui/MyNextUpView.java
package com.jwplayer.customui;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.lifecycle.LifecycleOwner;

import com.google.android.material.imageview.ShapeableImageView;
import com.jwplayer.pub.ui.viewmodels.NextUpViewModel;
import com.squareup.picasso.Picasso;

public class MyNextUpView extends ConstraintLayout {

    private TextView title;
    private TextView countdown;
    private ShapeableImageView poster;
    private View close;

    public MyNextUpView(Context context) {
        this(context, null);
    }

    public MyNextUpView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public MyNextUpView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, 0);
    }

    public MyNextUpView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initView(context);
    }

    private void initView(Context context) {
        inflate(context, R.layout.view_my_next_up, this);
        title = findViewById(R.id.next_title);
        countdown = findViewById(R.id.next_count);
        poster = findViewById(R.id.next_poster);
        close = findViewById(R.id.next_close);
    }

    public void bind(NextUpViewModel nextUpViewModel, LifecycleOwner lifecycleOwner) {
        nextUpViewModel.getThumbnailUrl().observe(lifecycleOwner, url -> {
            Picasso.get().load(url).into(poster);
        });
        nextUpViewModel.getTitle().observe(lifecycleOwner, string -> {
            title.setText(string);
        });
        nextUpViewModel.isUiLayerVisible().observe(lifecycleOwner, visible -> {
            setVisibility(visible ? VISIBLE : GONE);
        });
        nextUpViewModel.getNextUpTimeRemaining().observe(lifecycleOwner, remaining -> {
            countdown.setText("Next in: " + remaining);
        });
        close.setOnClickListener(v -> {
            nextUpViewModel.closeNextUpView();
        });
        setOnClickListener(v -> {
            nextUpViewModel.playNextPlaylistItem();
        });
    }

}

File: Kotlin CustomUi/app/src/main/java/com/jwplayer/customui/MyNextUpView.kt
package com.jwplayer.customui

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.lifecycle.LifecycleOwner
import com.google.android.material.imageview.ShapeableImageView
import com.jwplayer.pub.ui.viewmodels.NextUpViewModel
import com.squareup.picasso.Picasso

class MyNextUpView(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
  ConstraintLayout(
    context!!, attrs, defStyleAttr, defStyleRes
  ) {
  private var title: TextView? = null
  private var countdown: TextView? = null
  private var poster: ShapeableImageView? = null
  private var close: View? = null

  @JvmOverloads
  constructor(context: Context?, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : this(
    context,
    attrs,
    defStyleAttr,
    0
  )

  init {
    initView(context)
  }

  private fun initView(context: Context?) {
    inflate(context, R.layout.view_my_next_up, this)
    title = findViewById(R.id.next_title)
    countdown = findViewById(R.id.next_count)
    poster = findViewById(R.id.next_poster)
    close = findViewById(R.id.next_close)
  }

  fun bind(nextUpViewModel: NextUpViewModel, lifecycleOwner: LifecycleOwner?) {
    nextUpViewModel.thumbnailUrl.observe(lifecycleOwner!!) { url: String? ->
      Picasso.get().load(url).into(poster)
    }
    nextUpViewModel.title.observe(lifecycleOwner) { string: String? -> title!!.text = string }
    nextUpViewModel.isUiLayerVisible()
      .observe(lifecycleOwner) { visible: Boolean -> visibility = if (visible) VISIBLE else GONE }
    nextUpViewModel.nextUpTimeRemaining.observe(lifecycleOwner) { remaining: Int ->
      countdown!!.text = "Next in: $remaining"
    }
    close!!.setOnClickListener { v: View? -> nextUpViewModel.closeNextUpView() }
    setOnClickListener { v: View? -> nextUpViewModel.playNextPlaylistItem() }
  }
}
